test/core/test/repeats.test.ts

```diff
@@ -0,0 +1,43 @@
+import { afterAll, describe, expect, test } from 'vitest'
```

Comment:
Do you have any suggestions on how you think it should interact with "retries"? Because the way it is right now it will overlap retries.

Reply:
I feel that it should do the following:

- Test has `{ retries: 2, repeats: 5 }`
- Test fails
- Test reruns, `retries` decrements
- Test fails
- Test reruns, `retries` decrements
- Test fails
- Repeats decrements, retries are back to 2

So in the end it should run 15 times (1 for the actual run + 2 retries) x 5 retires

Reply:
`repeats`will run a test X amount of times, if the test fails during that loop, the loop breaks. `retries` will retry X amount of times and it doesn't care if the test fails during the loop, correct? So what should happen if we have both `repeats` and `retries` flags?

Reply:
> `repeats` will run a test X amount of times, if the test fails during that loop, the loop breaks. `retries` will retry X amount of times and it doesn't care if the test fails during the loop, correct?

The other way around. `repeats` (this PR adds a feature) runs how many tries it wants.

Reply:
> So what should happen if we have both repeats and retries flags?

each `repeat` has its own `retries`.

By default, `retries` is `0`, and `repeats` is `0`. So the test runs only one time.

If `retries` is `1` and `repeats` is `1`, the test will run one time. If it fails, it will run again. Then it will run again, because it should be "repeated". If it fails, it will run again, because `retries` is 1. So maximum reruns here is 4.

Reply:
I thought `repeats` would run until a test failed, since it's kinda of a stress test. But will change that behaviour then.

Reply:
Got it